src/DataGrid.tsx

```diff
@@ -378,6 +379,21 @@ function DataGrid<R, SR, K extends Key>(
     }
   });
 
+  useLayoutEffect(() => {
+    if (autoResizeColumn === null) return;
+    const columnElement = gridRef.current!.querySelector(
+      `[aria-colindex="${autoResizeColumn.idx + 1}"]`
+    )!;
+    const width = columnElement.clientWidth + 1;
```

Comment:
Not entirely sure why is this needed but may be because of right border. May be we can use gap to set the borders 🤔